scenarios/deployonly/build/build_application.go

```diff
@@ -1,43 +1,48 @@
 package build
 
 import (
-	models "github.com/equinor/radix-cicd-canary/generated-client/models"
+	"errors"
+	"fmt"
+
+	"github.com/equinor/radix-cicd-canary/generated-client/models"
 	"github.com/equinor/radix-cicd-canary/scenarios/utils/config"
 	envUtil "github.com/equinor/radix-cicd-canary/scenarios/utils/env"
 	httpUtils "github.com/equinor/radix-cicd-canary/scenarios/utils/http"
 	"github.com/equinor/radix-cicd-canary/scenarios/utils/job"
 	"github.com/equinor/radix-cicd-canary/scenarios/utils/test"
-	log "github.com/sirupsen/logrus"
 )
 
-var logger *log.Entry
-
 type expectedStep struct {
 	name       string
 	components []string
 }
 
 // Application Tests that we are able to successfully build an application
-func Application(env envUtil.Env, suiteName string) (bool, error) {
-	logger = log.WithFields(log.Fields{"Suite": suiteName})
+func Application(env envUtil.Env, suiteName string) error {
 
 	// Trigger build via web hook
-	ok, err := httpUtils.TriggerWebhookPush(env, config.App3BranchToBuildFrom, config.App3CommitID, config.App3SSHRepository, config.App3SharedSecret)
-	if !ok {
-		return false, err
+	err := httpUtils.TriggerWebhookPush(env, config.App3BranchToBuildFrom, config.App3CommitID, config.App3SSHRepository, config.App3SharedSecret)
+	if err != nil {
+		return fmt.Errorf("failed to push webhook push for App3, error %v", err)
 	}
 
 	// Get job
-	ok, jobSummary := test.WaitForCheckFuncOrTimeout(env, func(env envUtil.Env) (bool, interface{}) {
-		return job.IsListedWithStatus(env, config.App3Name, "Succeeded")
+	jobSummary, err := test.WaitForCheckFuncWithValueOrTimeout(env, func(env envUtil.Env) (*models.JobSummary, error) {
+		jobSummary, err := job.IsListedWithStatus(env, config.App3Name, "Succeeded")
+		if err != nil {
+			return nil, err
+		}
+		if jobSummary == nil {
+			return nil, fmt.Errorf("could not get listed job for application %s status '%s' - exiting", config.App3Name, "Succeeded")
+		}
```

Comment:
Should this if-block be moved out of the WaitForCheckFuncWithValueOrTimeout call? If a job reports Succeeded, but JobSummary is nil, there is no reason to retry the calls, and the Application function should just return with an error